mage_ai/frontend/api/utils/fetcher.ts

```diff
@@ -61,10 +61,10 @@ function preprocess(url: string, opts: FetcherOptionsType = {}) {
         type,
       } = file;
       const formData = new FormData();
-      const key: string = Object.keys(body).filter(k => k !== 'file')[0];
+      const bodyWithoutFile = Object.fromEntries(Object.entries(body).filter(([k]) => k !== 'file'));
```

Comment:
This line previously truncated all body keys, excluding only the first one, during file-related API calls. I discussed this matter with @johnson-mage last month, and he granted permission for the modification. As a result, all properties, including overwrite—currently employed by the API despite being a parameter in the createFile call—now pass through to the server. This adjustment was necessary to introduce a new flag, 'pipeline zip,' signaling to the server the specific nature of the operation.